verify.go

```diff
@@ -192,30 +215,46 @@ func VerifyIdentity(message string, audience string, getPubKey PubKeySource) (*a
 		return nil, fmt.Errorf("message identity mismatch. expected %q, got %q", audience, identity.Aud)
 	}
 
-	// TODO(miselin): need to check claims? and authority expiry?
-	return &identity, nil
-}
-
-func verifyClaims(iat time.Time, exp time.Time, replid string, claims *MessageClaims) error {
-	if iat.After(time.Now()) {
-		return fmt.Errorf("not valid for %s", time.Until(iat))
+	err = v.checkClaimsAgainstToken(&identity)
+	if err != nil {
+		return nil, fmt.Errorf("claim mismatch: %w", err)
 	}
 
-	if exp.Before(time.Now()) {
-		return fmt.Errorf("expired %s ago", time.Since(exp))
-	}
+	return &identity, nil
+}
 
+func verifyRawClaims(replid, user, cluster string, claims *MessageClaims) error {
 	if claims != nil {
-		authorized := false
+		if replid != "" {
+			if _, ok := claims.Repls[replid]; !ok {
```

Comment:
this also needs to have the check for the all-repls flag (same for the other two cases). can there also be a test for this case?

Reply:
Are you sure it needs that check? I don't think we _want_ all-repls at this level of verification.

Reply:
mostly for correctness: for identity verification purposes we'll never encounter this case since the per-repl certificate doesn't have that bit set ever. but it feels weird that one codepath checks that bit and the other one doesn't. (bonus points if both codepaths can be collapsed to the same one).

Reply:
This is handled now, test added to verify the "any *" flag case too.